### PR TITLE
Piston lifter prototype

### PR DESCRIPTION
**Pull request by Codadillo**
Code for the prototype of the support of the robot lifter.

### CODE REVIEW

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 1]
```diff
@@ -14,17 +16,27 @@
...
+		public static DoubleSolenoid liftPiston;
+	}
 	/**
 	 * The static initializer runs exactly once and ensures that
 	 * the variables are properly initialized.
 	 */
-	static {}
+	static {
+		//Lifter
+		Component.liftPiston = new DoubleSolenoid (Port.Pneumatics.liftSupportOut, Port.Pneumatics.liftSupportIn);
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Nit: extra space

File: src/org/usfirst/frc4904/robot/commands/RampLiftSupportOut.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	}
+	
+	@Override
+	protected void initialize() {
+		RobotMap.Component.lifter.extender.set(Lifter.SUPPORT_SOLENOID_RAISED);
+	}
+
+	@Override
+	protected boolean isFinished() {
+		// TODO Auto-generated method stub
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/commands/RampLiftSupportOut.java]: Take this out.

File: src/org/usfirst/frc4904/robot/commands/RampLiftSupportIn.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	}
+	
+	@Override
+	protected void initialize() {
+		RobotMap.Component.lifter.extender.set(Lifter.SUPPORT_SOLENOID_LOWERED);
+	}
+
+	@Override
+	protected boolean isFinished() {
+		// TODO Auto-generated method stub
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/commands/RampLiftSupportIn.java]: Remove comment.

File: src/org/usfirst/frc4904/robot/subsystems/Lifter.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+	public static final DoubleSolenoid.Value SUPPORT_SOLENOID_RAISED = DoubleSolenoid.Value.kReverse;
+	public static final DoubleSolenoid.Value SUPPORT_SOLENOID_LOWERED = DoubleSolenoid.Value.kForward;
+
+	public Lifter(DoubleSolenoid extender) {
+		this.extender = extender;
+	}
+
+	@Override
+	protected void initDefaultCommand() {
+		// TODO Auto-generated method stub
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/subsystems/Lifter.java]: What's the default command?

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 5]
```diff
@@ -14,17 +16,27 @@
 
 		public static class CAN {}
 
-		public static class Pneumatics {}
+		public static class Pneumatics {
+			public static int liftSupportOut = 0;
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Make these `-1` if they're unknown.

File: src/org/usfirst/frc4904/robot/subsystems/Lifter.java (hunk 4 above)
Comment (ajnadel) [reply src/org/usfirst/frc4904/robot/subsystems/Lifter.java]: (Probably `Idle` command)

**Review by ajnadel — CHANGES_REQUESTED** [review]
Get rid of the metadata junk.

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 6]
```diff
@@ -64,6 +74,7 @@
 			new VictorSP(Port.PWM.rightDriveA), new VictorSP(Port.PWM.rightDriveB));
 		Component.shifter = new SolenoidShifters(Port.Pneumatics.shifterUp, Port.Pneumatics.shifterDown);
 		Component.chassis = new TankDriveShifting("2018-Chassis", Component.leftWheel, Component.rightWheel, Component.shifter);
+
```
Comment (billwpierce) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: remove these extra added spaces

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 7]
```diff
@@ -53,6 +59,10 @@
 	 * the variables are properly initialized.
 	 */
 	static {
+
+		//Lifter
+		Component.liftPiston = new DoubleSolenoid(Port.Pneumatics.liftSupportOut, Port.Pneumatics.liftSupportIn);
```
Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Is there only one?

Comment (ajnadel) [inline src/org/usfirst/frc4904/robot/RobotMap.java]: Just initialize this in the `Lifter` constructor.

File: src/org/usfirst/frc4904/robot/RobotMap.java [hunk 8]
```diff
@@ -37,12 +39,25 @@
...
 		public static class Pneumatics {
+			// TODO: RealPorts
+			public static int leftLifterOut = -1;
+			public static int leftLifterIn = -1;
+			public static int rightLifterOut = -1;
+			public static int rightLifterIn = -1;
+			public static int leftLifterSupportIn = -1;
+			public static int leftLifterSupportOut = -1;
+			public static int rightLifterSupportIn = -1;
+			public static int rightLifterSupportOut = -1;
```
Comment (NikhilSuresh24) [inline src/org/usfirst/frc4904/robot/RobotMap.java:50]: Find out what these ports are

File: src/org/usfirst/frc4904/robot/commands/LifterSet.java [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+import edu.wpi.first.wpilibj.command.Command;
+
+public class LifterSet extends Command {
+	private DoubleSolenoid.Value state;
+	private DoubleSolenoid lifterPiston;
+
+	public LifterSet(DoubleSolenoid lifterPiston, DoubleSolenoid.Value state, Lifter lifter) {
+		this.lifterPiston = lifterPiston;
+		this.state = state;
+		requires(lifter);
```
Comment (NikhilSuresh24) [inline src/org/usfirst/frc4904/robot/commands/LifterSet.java]: put requires before other things

File: src/org/usfirst/frc4904/robot/humaninterface/operators/DefaultOperator.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+		RobotMap.Component.operatorStick.button3.onlyWhileHeld(new RunIf(
+			new LifterSupportOut(RobotMap.Component.lifterLeft),
+			RobotMap.Component.driverXbox.b::get,
+			() -> {
+				return RobotMap.Component.lifterLeft.extender.get() == Lifter.LIFTER_SOLENOID_RAISED;
+			}));
+
+		RobotMap.Component.operatorStick.button2.onlyWhileHeld(new RunIf(
+			new LifterOut(RobotMap.Component.lifterRight), RobotMap.Component.driverXbox.b::get));
+		RobotMap.Component.operatorStick.button4.onlyWhileHeld(new RunIf(
```
Comment (NikhilSuresh24) [inline src/org/usfirst/frc4904/robot/humaninterface/operators/DefaultOperator.java]: find out what these buttons should be

**Review by NikhilSuresh24 — CHANGES_REQUESTED** [review]

**Pull request merged**